low-level/solution/lr2/support_bundle/sspl_support_bundle

```diff
@@ -0,0 +1,369 @@
+            bundle_location = self._generate_tar(bundle_name)
+            SupportBundle._update_bundle_request_info(
+                bundle_id, request_id, bundle_location)
+            BundleRequestHandler._update_request_status(
+                bundle_id, BundleRequestHandler.SUCCESS)
+        except Exception as err:
+            status = BundleRequestHandler.FAILED + f" - {err}"
+            BundleRequestHandler._update_request_status(
+                bundle_id, status)
+
+    def get_sspl_bundle_data(self, bundle_id, request_id, file_path, include_encl):
+        """Collect SSPL data for support bundle."""
+        BundleRequestHandler._add_bundle_request(bundle_id, request_id, file_path)
+
+        if file_path:
+            if os.path.exists(file_path):
+                file_path = file_path+"/sspl/"
+                self.tmp_bundle_path = os.path.normpath(file_path)
+            else:
+                msg = "Given path %s doesn't exist" % (file_path)
+                self.__clear_tmp_files()
+                raise SupportBundleError(1, msg)
+        os.makedirs(self.tmp_bundle_path, exist_ok=True)
+        try:
+            self.get_ipmi_sel_data()
+            self.get_config_data()
+            self.get_drives_smart_data()
+            if not include_encl:
+                RealStorEnclosure().get_enclosure_logs(
+                    self.enclosure_log, logger)
+
+            if os.path.exists(self.enclosure_log) and not include_encl:
+                enclosure_zip_file = "enclosure-wwn-{0}-logs-{1}.zip".format(
+                    self.enclosure_wwn, str(int(time.time())))
+                shutil.copy(self.enclosure_log,
+                            self.tmp_bundle_path+enclosure_zip_file)
+                logger.info("Enclosure Log File Location: %s" %
+                            self.tmp_bundle_path+enclosure_zip_file)
+        except OSError as err:
+            msg = "Facing problem while creating sspl "
+            "support bundle: {0}".format(err)
+            self.__clear_tmp_files()
+            raise SupportBundleError(1, msg)
+
+    def get_ipmi_sel_data(self):
+        ipmitool = IpmiFactory().get_implementor('ipmitool')
+        res, err, retcode = ipmitool._run_ipmitool_subcommand("sel list")
+        if retcode != 0:
+            if err.find(ipmitool.VM_ERROR) != -1:
+                err = ("Detected VM environment, can't reach BMC over inband"
+                       "KCS channel, avoiding BMC data collection using IPMI.")
+            logger.warn(
+                f"Unable to fetch SEL list data with an error : {err}")
+        else:
+            with open(self.ipmi_sel_data, 'w+') as sel_list:
+                sel_list.write(res)
+
+    def get_config_data(self):
+        os.makedirs(self.sspl_conf_dir, exist_ok=True)
+        for conf_file in glob.glob('/etc/sspl*.*'):
+            shutil.copy(conf_file, self.sspl_conf_dir)
+
+    def get_drives_smart_data(self):
```

Comment:
why this function still here then, when moved to DriveUtils class, from where it should be used

Reply:
It is not a generic function. here we are just dumping local drive data into a file and adding this file into SB. So it is SB-specific code only.